diff -urN a/src/file.c b/src/file.c
--- a/src/file.c	2015-04-01 10:25:11.000000000 +0100
+++ b/src/file.c	2015-04-02 00:24:04.000000000 +0100
@@ -227,8 +227,10 @@
 
 #ifdef TESTFILE
 int map_w, map_h;	/* shut up linker in misc.c */
-int char_width() { return 0;}
-int parser_get_string() { return 0;}
+typedef struct _Font Font;
+int char_width(Font *fnt, char c) { (void)fnt; (void)c; return 0;}
+typedef struct _PData PData;
+int parser_get_string(PData *pd, char *name, char   **result) { (void)pd; (void)name; (void)result; return 0;}
 
 int main( int argc, char **argv )
 {
