diff -urN a/unicon/ImmModules/turbo/levelphrase.c b/unicon/ImmModules/turbo/levelphrase.c
--- a/unicon/ImmModules/turbo/levelphrase.c	2015-08-16 13:59:40.000000000 +0100
+++ b/unicon/ImmModules/turbo/levelphrase.c	2015-08-16 14:00:24.000000000 +0100
@@ -169,7 +169,7 @@
     FILE *fp;
     int i;
     fp = fopen (szFileName, "wt");
-    fprintf (fp, "PhraseLevel[256] = { \n    ");
+    fprintf (fp, "int PhraseLevel[256] = { \n    ");
     for (i = 0; i  < 256; i++)
     {
         if (i % 16 == 0)
diff -urN a/unicon/ImmModules/turbo/test.h b/unicon/ImmModules/turbo/test.h
--- a/unicon/ImmModules/turbo/test.h	2015-08-16 13:57:04.000000000 +0100
+++ b/unicon/ImmModules/turbo/test.h	2015-08-16 13:59:48.000000000 +0100
@@ -1,4 +1,4 @@
-PhraseLevel[256] = { 
+int PhraseLevel[256] = { 
     
     0,30,60,90,120,150,180,210,240,270,300,330,360,390,420,450,
     480,510,540,570,600,630,660,690,720,750,780,810,840,870,900,930,
