From cea4894deb02e480c3571de1868fb425d38c6d9a Mon Sep 17 00:00:00 2001
From: jschpp <mail@jschpp.de>
Date: Sat, 20 Feb 2016 16:02:43 +0100
Subject: [PATCH] Fix #845

---
 src/search.h | 1 +
 src/util.h   | 8 --------
 2 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/src/search.h b/src/search.h
index aadf1fa..d4f9c08 100644
--- a/src/search.h
+++ b/src/search.h
@@ -28,6 +28,7 @@
 #include "options.h"
 #include "print.h"
 #include "util.h"
+#include "decompress.h"
 #include "uthash.h"
 
 size_t alpha_skip_lookup[256];
diff --git a/src/util.h b/src/util.h
index cba1306..0d84592 100644
--- a/src/util.h
+++ b/src/util.h
@@ -42,12 +42,6 @@ typedef struct {
     struct timeval time_end;
 } ag_stats;
 
-typedef enum {
-    AG_NO_COMPRESSION,
-    AG_GZIP,
-    AG_COMPRESS,
-    AG_ZIP
-} ag_compression_type;
 
 ag_stats stats;
 
@@ -74,8 +68,6 @@ size_t invert_matches(const char *buf, const size_t buf_len, match_t matches[],
 void realloc_matches(match_t **matches, size_t *matches_size, size_t matches_len);
 void compile_study(pcre **re, pcre_extra **re_extra, char *q, const int pcre_opts, const int study_opts);
 
-void *decompress(const ag_compression_type zip_type, const void *buf, const int buf_len, const char *dir_full_path, int *new_buf_len);
-ag_compression_type is_zipped(const void *buf, const int buf_len);
 
 int is_binary(const void *buf, const size_t buf_len);
 int is_regex(const char *query);
