--- a/core/libsparse/sparse_crc32.c
+++ b/core/libsparse/sparse_crc32.c
@@ -98,7 +98,7 @@
  * in sys/libkern.h, where it can be inlined.
  */
 
-uint32_t sparse_crc32(uint32_t crc_in, const void *buf, int size)
+uint32_t sparse_crc32(uint32_t crc_in, const void *buf, unsigned int size)
 {
         const uint8_t *p = buf;
         uint32_t crc;
--- a/core/libsparse/sparse_read.c
+++ b/core/libsparse/sparse_read.c
@@ -102,7 +102,7 @@
 		uint32_t *crc32)
 {
 	int ret;
-	int chunk;
+	unsigned int chunk;
 	unsigned int len = blocks * s->block_size;
 
 	if (chunk_size % s->block_size != 0) {
@@ -139,7 +139,7 @@
 		int fd, unsigned int blocks, unsigned int block, uint32_t *crc32)
 {
 	int ret;
-	int chunk;
+	unsigned int chunk;
 	int64_t len = (int64_t)blocks * s->block_size;
 	uint32_t fill_val;
 	uint32_t *fillbuf;
@@ -189,7 +189,7 @@
 		memset(copybuf, 0, COPY_BUF_SIZE);
 
 		while (len) {
-			int chunk = min(len, COPY_BUF_SIZE);
+			unsigned int chunk = min(len, COPY_BUF_SIZE);
 			*crc32 = sparse_crc32(*crc32, copybuf, chunk);
 			len -= chunk;
 		}
--- a/core/libsparse/sparse_crc32.h
+++ b/core/libsparse/sparse_crc32.h
@@ -16,5 +16,4 @@
 
 #include <stdint.h>
 
-uint32_t sparse_crc32(uint32_t crc, const void *buf, size_t size);
-
+uint32_t sparse_crc32(uint32_t crc, const void *buf, unsigned int size);
