diff -urN a/umac.c b/umac.c
--- a/umac.c	2016-02-20 00:16:38.000000000 +0000
+++ b/umac.c	2016-02-20 00:17:06.000000000 +0000
@@ -1179,7 +1179,7 @@
     uhash_ctx hash;          /* Hash function for message compression    */
     pdf_ctx pdf;             /* PDF for hashed output                    */
     void *free_ptr;          /* Address to free this struct via          */
-} umac_ctx;
+};
 
 /* ---------------------------------------------------------------------- */
 
